### ISSUE TITLE
GQL Base Interface for Link field

### ISSUE THREAD

**Issue by r0skar** [verbb/hyper#69]
### Question

Hi,

Is it possible to add a general/base interface for the link field type similar to how the TypedLink and LinkIt plugins do it? Not sure I am using the right wording here, so let me show an example. At the moment I create a fragment for a link like this:

```gql
fragment CtaLink on callToActionLink_Link {
  text
  url
}
```

However, since all links are basically the same, I use a generic Link component in my frontend. Ideally, I could create my fragment there using the base gql type:

```gql
fragment Link on Link_Link {
  text
  url
}
```

**Comment by engram-design** [comment]
All link objects implement the [LinkInterface](https://verbb.io/craft-plugins/hyper/docs/developers/graphql#the-linkInterface-interface) if that’s what you mean?

`fragment Link on LinkInterface` should be suitable?

**Comment by r0skar** [comment]
Yes that's what I am looking for, but it seems the interface is not registered(?)
![image](https://github.com/verbb/hyper/assets/33655345/89e16487-b676-41e8-83f5-e2cf0eae662c)

**Comment by engram-design** [comment]
Ah, my mistake. Should be fixed for the next release. To get this early run `composer require verbb/hyper:"dev-craft-4 as 1.1.7"`

```gql
fragment CtaLink on HyperLinkInterface {
  text
  url
}

{
  entries(id: 123) {
    hyperField {
      ... CtaLink
    }
  }
}
```

**Comment by engram-design** [comment]
Fixed in [1.1.8](https://github.com/verbb/hyper/releases/tag/1.1.8)
